### PR TITLE
Add public SSH key selection

### PR DESCRIPTION
**Pull request by ezr-ondrej**
I've added the `uploadedKey` to the store and changed the validated step.

Error state:
image

### CODE REVIEW

File: src/Components/ExpandableAWS/index.js
Comment (adiabramovitch) [inline src/Components/ExpandableAWS/index.js:6]: Does that mean that AWS default is open?

Comment (ezr-ondrej) [reply src/Components/ExpandableAWS/index.js:6]: Yes, is that an issue? :thinking: I thought it is better for the time being (until we have other providers)

**Comment by adiabramovitch** [comment]
Do we want to add a test?

**Comment by adiabramovitch** [comment]
And maybe we should change it to SelectOption like instance type? if we want to be consistent

**Comment by ezr-ondrej** [comment]
> Do we want to add a test?

Dunno, IMHO this does nothing and @amirfefer was voicing against too much testing on client, but I'm fine with whatever, I can add test once touching it to change to Select

> SelectOption like instance type

Yes, once that is in, I'll update this one to comply :)

File: src/Components/Common/WizardContext/initialState.js
Comment (amirfefer) [inline src/Components/Common/WizardContext/initialState.js:5]: is this flag needed? if the `chosenSshKeyId` is defined we can deduce it anyway, no?

File: src/Components/Wizard/steps/index.js
Comment (amirfefer) [inline src/Components/Wizard/steps/index.js:40]: nice :)

File: src/Components/Common/WizardContext/initialState.js
Comment (amirfefer) [inline src/Components/Common/WizardContext/initialState.js:9]: why not use `sshPublicName`?

**Review by amirfefer — COMMENTED** [review]
Thanks, @ezr-ondrej  👍  looks good
I left some minor comments

File: src/Components/Common/WizardContext/initialState.js
Comment (ezr-ondrej) [reply src/Components/Common/WizardContext/initialState.js:5]: I don't think so, what happens if you choose a key, but then change your mind and upload it instead, we will have `chosenSshKeyId` defined, won't we?

File: src/Components/Common/WizardContext/initialState.js
Comment (ezr-ondrej) [reply src/Components/Common/WizardContext/initialState.js:9]: the same reason as above, they could collide, although here we could probably fix that by nullifying the vars on switching, but then you'd loose what you've filled out when you:
1. go to upload a key, you fill out everyting
2. you wanna check you do not have some key that you remembered you might have
3. you switch to a choose public key
4. you go throught them, select some by accident
5. you go back to uploading

IMHO in this scenario the public key should not be gone. I'll write tests for that.

Comment (ezr-ondrej) [reply src/Components/Common/WizardContext/initialState.js:9]: also, I think that `sshPublicName` should be renamed at some point, but that is irrelevant to this discussion :)

File: src/Components/Common/WizardContext/initialState.js
Comment (amirfefer) [reply src/Components/Common/WizardContext/initialState.js:5]: we can just reset the `chosenSshKeyId` back to undefined when the user uploads a key

File: src/Components/Common/WizardContext/initialState.js
Comment (amirfefer) [reply src/Components/Common/WizardContext/initialState.js:9]: as you mentioned, this scenario can be fixed by just resetting the state on switching. I'd prefer fewer states, especially when the state is shareable among the whole wizard though I won't block this PR on this, we can improve this later on (i.e better naming `sshPublicName` ->  `uploadedSSHKeyName`)

**Review by amirfefer — COMMENTED** [review]
Tested and works well, Thanks @ezr-ondrej !
Before merging, what should be the default option? according to the mockups looks like the upload a key is the default option, but I'm not sure

**Comment by ezr-ondrej** [comment]
I believe that selecting is the default, which makes sense, as most users will probably use one (or limited set) key everytime for all the company hosts.
But good question, because it got me thinking about situation you have no keys uploaded, in which case it might be better to default to the uploading.

**Comment by ezr-ondrej** [comment]
Tests added and transformed to using Select component :)

File: src/Components/Common/WizardContext/initialState.js
Comment (ezr-ondrej) [reply src/Components/Common/WizardContext/initialState.js:9]: I've added test for the scenario I've mentioned above, I believe simple reuse of the same state would break the test now.
Take a look at the test if you agree please :)

File: src/Components/Common/WizardContext/initialState.js
Comment (ezr-ondrej) [reply src/Components/Common/WizardContext/initialState.js:5]: we can for sure, I just thought that would be much more corner cases to worry about, it is doable, but this felt bit more straight forward :shrug:

**Review by amirfefer — APPROVED** [review]
Thanks, @ezr-ondrej ! 
Let's merge!

**Pull request merged**
